compiler/compile/OSRData.cpp

```diff
@@ -513,6 +514,388 @@ uint32_t TR_OSRCompilationData::getOSRStackFrameSize(uint32_t methodIndex)
+ * method. Note that it's possible that the immediate next caller doesn't have an OSRCodeBlock
+ * and the `next OSRCodeBlock` would belong to the caller's caller.
+ *
+ * If called from `buildDefiningMap`, the map is for the OSRCatchBlock, and the `next OSRCodeBlock`
+ * belongs to the same method as the OSRCatchBlock.
+ */
+void
+TR_OSRMethodData::buildDefiningMapForBlock(TR::Block *block, DefiningMap *blockMap)
+   {
+   TR_ASSERT(block->getSuccessors().size() == 1, "OSRCatchBlock should have one successor but block_%d has %d", block->getNumber(), block->getSuccessors().size());
+   if (comp()->getOption(TR_TraceOSR))
+      traceMsg(comp(), "buildDefiningMapForBlock block_%d\n", block->getNumber()); 
+   buildDefiningMap(block, blockMap);
+   TR::Block *nextBlock = toBlock(block->getSuccessors().front()->getTo()); 
+   if (!nextBlock->isOSRCodeBlock())
+      buildDefiningMapForBlock(nextBlock, blockMap);
+   }
+
+/*
+ * \brief This is the most basic function iterating through all store nodes and add entries to the DefiningMap to
+ * map each symbol to the group of symbols defining it
+ */
+void
+TR_OSRMethodData::buildDefiningMap(TR::Block *block, DefiningMap *blockMap, DefiningMap *prepareForOSRCallMap)
+   {
+   for (TR::TreeTop *tt = block->getEntry(); tt != block->getExit(); tt = tt->getNextRealTreeTop())
+      {
+      TR::SymbolReference * symRef = NULL;
+      TR::Node *node = tt->getNode();
+
+      if (comp()->getOption(TR_TraceOSR))
+         traceMsg(comp(), "buildDefiningMap node n%dn\n", node->getGlobalIndex());
+
+      if (node->getOpCode().isStoreDirect())
+         symRef = node->getSymbolReference();
+      else if (node->getOpCode().isStoreReg())
+         symRef = node->getRegLoadStoreSymbolReference();
+
+      if (symRef)
+         {
+         if (symRef->getSymbol()->isAutoOrParm())
+            {
+            TR_BitVector subTreeSymRefs;
+            TR::NodeChecklist checklist(comp());
+            collectSubTreeSymRefs(node->getFirstChild(), &subTreeSymRefs, checklist);
+
+            if (comp()->getOption(TR_TraceOSR))
+               {
+               traceMsg(comp(), "buildDefiningMap: node n%dn: defining symbol #%d: ", node->getGlobalIndex(), symRef->getReferenceNumber());
+               subTreeSymRefs.print(comp());
+               traceMsg(comp(), "\n");
+               }
+
+            TR_BitVector *symRefs = NULL;
+            if (blockMap->find(symRef->getReferenceNumber()) != blockMap->end())
+               {
+               symRefs = (*blockMap)[symRef->getReferenceNumber()];
+               symRefs->empty(); // the current store kills previous defs
+               }
+
+            if (!subTreeSymRefs.isEmpty())
+               {
+               if (!symRefs)
+                  symRefs = new (comp()->trStackMemory()) TR_BitVector(0, comp()->trMemory(), stackAlloc);
+               updateDefiningSymRefs(&subTreeSymRefs, blockMap, symRefs);
```

Comment:
`symRefs` is extracted from the current map if there is already a mapping for the `symRef`, otherwise its allocated here. In the event its allocated, its never added back to the map. Is this intentional? If so, why allocate and run `updateDefiningSymRefs` at all?

Reply:
You are right. The new allocated symRefs should be added back to the map. Nice Catch!